Dev

SlavZ27 opened the pull request.

Commit:
Fix API description: fix "ParameterIn" from DEFAULT to QUERY:
AuthController, UsersApiController

Commit:
Merge branch 'dev' into feature-Zaytsev

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/AdsApiController.java
#	src/main/java/ru/skypro/homework/controller/api/AdsApi.java

nadillustrator committed:
Create repositories

nadillustrator committed:
Create liquibase dependencies in pom.xml & structure in resources

nadillustrator committed:
Create liquibase dependencies in pom.xml, structure in resources, migrations for creating tables
Some fixes for Entities & Repositories

Commit:
Merge branch 'dev' into feature-Zaytsev

evnag committed:
Merge pull request #9 from SlavZ27/feature_nadillustrator_2-create_DB_with_liquibase

Feature nadillustrator 2 create db with liquibase

Commit:
Merge branch 'dev' into feature-Zaytsev

Commit:
feat: added mapstruct dependency&plugins
refactor: change type of price (add new migration)
refactor: changed the type of price to an Integer
refactor: removed initialization of fields with nulls(CreateAdsDto,CommentDto)
refactor: removed city field from UserDto

nadillustrator committed:
Correct entity and schema Comment

evnag committed:
Merge pull request #10 from SlavZ27/feature_nadillustrator_correct_comment

Correct entity and schema Comment

Commit:
Merge branch 'dev' into feature-Zaytsev

Commit:
Merge branch 'dev' into feature-evnag-07-02-23

# Conflicts:
#	src/main/resources/liquibase/scripts/tasks-log.sql

Commit:
feat: added mappers
feat: added tests for mappers

nadillustrator committed:
Merge pull request #11 from SlavZ27/feature-evnag-07-02-23

Add mappers, tests; Add some fixes; evnag 07 02 23

Commit:
Merge branch 'dev' into feature-Zaytsev

Commit:
add AdsServiceImpl
add CommentServiceImpl
add ImageServiceImpl
add UserServiceImpl

nadillustrator committed:
Merge pull request #12 from SlavZ27/feature-Zaytsev

add services

Commit:
add Faker to pom.xml
add AdsApiControllerTest
add Generator for tests
add generation data for DB to AdsApiControllerTest
add first test (contextLoads) to AdsApiControllerTest

nadillustrator committed:
Correct entity and schema Avatar

evnag committed:
Merge pull request #14 from SlavZ27/feature_nadillustrator_correct_avatar

Correct Avatar entity and schema

nadillustrator committed:
Correct entities toString, pom

nadillustrator committed:
Merge pull request #13 from SlavZ27/feature-Zaytsev

add first test (contextLoads) to AdsApiControllerTest

nadillustrator committed:
Create Ads delete method

Exesebaf committed:
Merge pull request #15 from SlavZ27/feature_nadillustrator_entities_correct_tostring

Correct entities toString, pom

nadillustrator committed:
Create Ads delete method
Delete @NoArgsConstructor annotations

Commit:
fix generate Image from Generator
add images for tests
add method getImage to ImageApiController
add method getImage to ImageServiceImpl
add positive and negative tests to ImageApiControllerTest
add ImageNotFoundException

SlavZ27 committed:
Merge pull request #16 from SlavZ27/feature_nadillustrator_create_andpoints_delete

Feature nadillustrator create andpoints delete

Commit:
Merge branch 'dev' into feature-Zaytsev

nadillustrator committed:
Merge pull request #17 from SlavZ27/feature-Zaytsev

process endpoint GET localhost:8080/image/{idImage}

Commit:
fix method removeAds to AdsApiController
fix method removeAds to AdsServiceImpl
add methods findAllByIdAds to CommentRepository and ImageRepository
add methods to ImageServiceImpl and CommentServiceImpl
add positive and negative tests to AdsApiControllerTest

evnag committed:
Merge pull request #18 from SlavZ27/feature-Zaytsev

process endpoint DELETE localhost:8080/ads/{id}

Commit:
feat: added processing POST localhost:8080/ads/{id}
feat: added processing POST localhost:8080/ads/{ads_pk}/comments
feat: added processing POST localhost:8080/ads/{ads_pk}/comments/{id}
feat: added processing GET localhost:8080/ads/{ads_pk}/comments/{id}
feat: added processing GET localhost:8080/ads/{ads_pk}/comments
feat: added mapping logic to ResponseWrapperCommentDto in the CommentMapper class
feat: added CommentNotFoundException
feat: added http-client dependency to pom.xml for PATCH tests

Commit:
feat: added processing POST localhost:8080/ads
refactor: change parameters of few methods(String adPk to Integer adPk)
refactor: CommentNotFoundException, AdsNotFoundException, AuthorNotFoundException

SlavZ27 committed:
Merge pull request #19 from SlavZ27/feature-evnag-07-02-23

Added processing of multiple endpoints

Exesebaf committed:
added getAds(Integer idAds), getALLAds()  to AdsServiceImpl and AdsApiController
added mapper for ResponseWrapperAdsDto to AdsMapper
FullAdsMapper, ImageServiceImpl, application.properties added a tie to the address of the images.

evnag committed:
Merge pull request #20 from SlavZ27/feature-Aleksander_Vavilov

Cool pullrequest

Commit:
path.to.avatars.folder=./avatars
path.to.avatars.folder=./avatars_test
add AvatarNotFoundException
add AvatarServiceImpl and methods
fix generate avatar to Generator
add updateImage to ImageServiceImpl
add getAvatar to UsersApiController
Fix methos for UsersApiController
add UserServiceImpl and methods

nadillustrator committed:
Merge pull request #21 from SlavZ27/feature-Zaytsev

More changes

Commit:
feat: added processing DELETE localhost:8080/ads/{ads_pk}/comments/{id}

SlavZ27 committed:
Merge pull request #22 from SlavZ27/feature-evnag-07-02-23

Added processing for DELETE localhost:8080/ads/{ads_pk}/comments/{id}

Commit:
Fix mapping and fix getLink for images

nadillustrator committed:
Merge pull request #23 from SlavZ27/feature-Zaytsev

Fix mapping and fix getLink for images

nadillustrator committed:
Refactoring api interfaces -> classes ApiControllers

nadillustrator committed:
OneToOne for Ads -> Image
Add id_image to entity Ads
Fix DB for Ads+Image using liquibase script
Fix method addAds class AdsServiceImpl
Fix methods addImage ImageServiceImpl

nadillustrator committed:
Fix List<Image> images ->Image image (AdsMapper, FullAdsDto)

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsApiController.java:
```diff
@@ -5,165 +5,83 @@
 
     private final ObjectMapper objectMapper;
-
+    private final AdsServiceImpl adsServiceImpl;
```
если нет интерфейсов, то классы лучше переименовать без Impl на конце

SlavZ27 replied:
Была надежда сделать интерфейсы

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsApiController.java:
```diff
@@ -5,165 +5,83 @@
     private final ObjectMapper objectMapper;
-
+    private final AdsServiceImpl adsServiceImpl;
     private final HttpServletRequest request;
```
это не нужно после рефакторинга, насколько я вижу

SlavZ27 replied:
Немного не досмотрели

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsApiController.java:
```diff
@@ -5,165 +5,83 @@
+
+    public ResponseEntity<AdsDto> addAds(
+            @Parameter(in = ParameterIn.DEFAULT, description = "", schema = @Schema()) @RequestPart(value = "properties", required = false) CreateAdsDto properties,
+            @Parameter(description = "file detail") @Valid @RequestPart("image") MultipartFile image) throws IOException {
```
а как файл можно провалидировать?)

SlavZ27 replied:
Наверное никак

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/mapperService.java:
```diff
@@ -0,0 +1,15 @@
+import java.util.List;
+
+@Service
+public class mapperService {
```
пустой класс

SlavZ27 replied:
Недосмотр

dmitry-bizin commented on src/test/java/ru/skypro/homework/Generator.java:
```diff
@@ -0,0 +1,360 @@
+
+public class Generator {
+    private final Faker faker = new Faker();
+    private final Random random = new Random();
```
у фейкера в принципе есть рандом внутри

SlavZ27 replied:
Просто было удобнее с рандомом

dmitry-bizin commented on src/test/java/ru/skypro/homework/controller/AdsApiControllerTest.java:
```diff
@@ -0,0 +1,321 @@
+                .filter(ads -> commentRepository.findAllByIdAds(ads.getId()).size() > 0 &&
+                        imageRepository.findAllByIdAds(ads.getId()).size() > 0)
+                .findAny().orElse(null);
+        assert actualAds != null;
```
а что не assertj-ыми ассёртами?)

SlavZ27 replied:
Так не ругается системы на вызов геттеров у объекта

dmitry-bizin replied:
можно через assertJ проверить сначала на не налл, а потом уже вызывать геттеры, тогда не будет ругаться)

nadillustrator committed:
Fix Method Parameters Ads, User -> idAds, idUser

nadillustrator committed:
Half of work with tests

nadillustrator committed:
Revert Fix Method Parameters Ads, User -> idAds, idUser

nadillustrator committed:
getAdsMeUsingGET

evnag committed:
Merge pull request #25 from SlavZ27/feature_nadillustrator_refactoring_2week

Feature nadillustrator refactoring 2week

Commit:
Fix project with recommendation of mentor

nadillustrator committed:
Merge pull request #26 from SlavZ27/feature-Zaytsev

Fix project with recommendation of mentor

nadillustrator committed:
Add local date to Ads
Fix DB

nadillustrator committed:
Move methods from ImageController to AdsController

Commit:
Make one standard user (User) for modified
Remake Image and ImageService to work without Ads
Remake Avatar and AvatarService to work without User

evnag committed:
Merge pull request #27 from SlavZ27/feature-nadillustrator-add_localdatetime_to_Ads

Feature nadillustrator add localdatetime to ads

Commit:
Merge branch 'dev' into feature-Zaytsev

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/ImageApiController.java
#	src/main/resources/liquibase/scripts/tasks-log.sql

Commit:
Make one standard user (User) for modified
Remake Image and ImageService to work without Ads
Remake Avatar and AvatarService to work without User

nadillustrator committed:
Merge pull request #28 from SlavZ27/feature-Zaytsev

Feature zaytsev

Commit:
feat: added processing for GET localhost:8080/ads/by-title

Commit:
refactor: removed an unused method from AdsMapper

nadillustrator committed:
Merge pull request #29 from SlavZ27/feature-evnag-07-02-23

feat: added processing for GET localhost:8080/ads/by-title
